.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,98 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '34 4 * * 1'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'csharp' ]
+        framework: [netcoreapp3.1, net5.0]
+        consul: [1.6.10, 1.7.14, 1.8.13, 1.9.7, 1.10.0]
```

Comment:
I think there is no need to use a matrix for `framework` and `consul`. `framework` is used in the main CI pipeline to parallelise jobs, and `consul` is used to determine which version of the Consul agent to run the tests against. Here we don't run the tests, but instead we just want to build the entire solution for all frameworks (for which autobuild should actually be enough I guess).